jobs/build-and-deploy-publish-worker.groovy

```diff
@@ -0,0 +1,107 @@
+// Pipeline job that creates a new dockerfile image for the publish-worker,
+// and uploads it to our docker container-registry.
+
+@Library("kautils")
+// Classes we use, under jenkins-jobs/src/.
+import org.khanacademy.Setup;
+// Vars we use, under jenkins-jobs/vars/.  This is just for documentation.
+//import vars.kaGit
+//import vars.notify
+//import vars.onWorker
+
+
+new Setup(steps
+
+).addStringParam(
+    "GIT_REVISION",
+    """The name of a webapp branch to use when building the docker image.
+Most of the time master (the default) is the correct choice. But it can
+be any commit that is "later" than the one for the last publish worker.""",
+    "master"
+
+).addStringParam(
+    "ZND_NAME",
+    """If you specified the commit of a znd in GIT_REVISION, put the
+znd's name here.  It will be appended to the docker image name, to
+indicate that this image does not derive from prod.""",
+    ""
+
+).addBooleanParam(
+    "VALIDATE_COMMIT",
+    """If set, check that this commit is safe to publish.  <b>UNSET
+WITH CAUTION!</b> -- only for situations where you are overwriting
+someone else's publish-deploy on purpose.""",
+    true
+
+).addBooleanParam(
+    "DEPLOY_TO_PROD",
+    """If set, actually deploy the new publish-worker image to production. If not set, the job will stop after building the image.""",
+    false
+
+).addStringParam(
+    "DEPLOYER_EMAIL",
+    """The email address of the deployer. This will be used for deployment tracking and auditing purposes.""",
+    ""
+	
+).apply();
+
+currentBuild.displayName = "${currentBuild.displayName} (${params.GIT_REVISION})";
+
+def _setupWebapp() {
+   withTimeout('1h') {
+   		kaGit.safeSyncToOrigin("git@github.com:Khan/webapp", 
+			params.GIT_REVISION);
```

Comment:
Style nit: the indentation here is weird, because you're using tabs.  Make sure that you're only using spaces in this file.